File: VB/Helpers/MyScrollBarPaintHelper.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.LookAndFeel
Imports DevExpress.Utils.Drawing
Imports DevExpress.Utils
Imports System
Imports System.Drawing
Namespace WindowsApplication1
	Public NotInheritable Class MyScrollBarPaintHelper
		Private Shared _ScrollBar As VScrollBar = New MyScrollBar()
		Private Shared _Painter As ScrollBarPainterBase
	   Private Sub New()
	   End Sub
	   Shared Sub New()
			CreatePainter()
			AddHandler UserLookAndFeel.Default.StyleChanged, AddressOf Default_StyleChanged
	   End Sub
		Private Shared Sub Default_StyleChanged(ByVal sender As Object, ByVal e As EventArgs)
			CreatePainter()
		End Sub
		Private Shared Sub CreatePainter()
			_Painter = LookAndFeelScrollBarPainter.GetPainter(UserLookAndFeel.Default)
		End Sub
		Public Shared Sub DrawScrollBar(ByVal cache As GraphicsCache, ByVal bounds As Rectangle)
			bounds.X = bounds.Right - _ScrollBar.Width - 1
			bounds.Width = _ScrollBar.Width
			_ScrollBar.Height = bounds.Height
			Dim args As New ScrollBarInfoArgs(cache, bounds, AppearanceObject.ControlAppearance, AppearanceObject.ControlAppearance, _ScrollBar)
			TryCast(args.ViewInfo, MyScrollBarViewInfo).Reset()
			TryCast(args.ViewInfo, MyScrollBarViewInfo).ScrollBarOffset = bounds.Location
			_Painter.DrawObject(args)
		End Sub
	End Class


End Namespace

File: CS/Helpers/MyScrollBarPaintHelper.cs
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.LookAndFeel;
using DevExpress.Utils.Drawing;
using DevExpress.Utils;
using System;
using System.Drawing;
namespace WindowsApplication1
{
    public static class MyScrollBarPaintHelper
    {
        private static VScrollBar _ScrollBar = new MyScrollBar();
        private static  ScrollBarPainterBase _Painter;
       static MyScrollBarPaintHelper()
        {
            CreatePainter();
            UserLookAndFeel.Default.StyleChanged += Default_StyleChanged;
        }
        static  void Default_StyleChanged(object sender, EventArgs e)
        {
            CreatePainter();
        }
        private static void CreatePainter()
        {
            _Painter = LookAndFeelScrollBarPainter.GetPainter(UserLookAndFeel.Default);
        }
        public static  void DrawScrollBar(GraphicsCache cache, Rectangle bounds)
        {
            bounds.X = bounds.Right - _ScrollBar.Width - 1;
            bounds.Width = _ScrollBar.Width;
            _ScrollBar.Height = bounds.Height;
            ScrollBarInfoArgs args = new ScrollBarInfoArgs(cache, bounds, AppearanceObject.ControlAppearance, AppearanceObject.ControlAppearance, _ScrollBar);
            (args.ViewInfo as MyScrollBarViewInfo).Reset();
            (args.ViewInfo as MyScrollBarViewInfo).ScrollBarOffset = bounds.Location;
            _Painter.DrawObject(args);
        }
    }


}